fake/fake_clock_test.go

```diff
@@ -668,3 +668,90 @@ func TestFakeClockAfterFuncNegDur(t *testing.T) {
 	}
 
 }
+
+func TestFakeClockContext(t *testing.T) {
+	t.Run("ContextDeadline", func(t *testing.T) {
+		base := time.Now()
+		c := NewClock(base)
+
+		deadline := base.Add(1)
+		ctx, cancel := c.ContextWithDeadline(context.Background(), deadline)
+		t.Cleanup(cancel)
+
+		ctxDeadline, isSet := ctx.Deadline()
+		if !isSet {
+			t.Errorf("context deadline not set")
+		}
+		if !ctxDeadline.Equal(deadline) {
+			t.Errorf("unexpected context deadline: %v; expected %v", ctxDeadline, deadline)
+		}
+	})
+
+	t.Run("ContextWithDeadlineExceeded", func(t *testing.T) {
+		base := time.Now()
+		c := NewClock(base)
+
+		ctx, cancel := c.ContextWithDeadline(context.Background(), base.Add(1))
+		t.Cleanup(cancel)
+
+		c.Advance(1)
```

Comment:
I think it makes sense to verify that `Advance` returns `1` -- indicating that it ran one callback (or awoke a goroutine).